tephra-core/src/main/java/co/cask/tephra/coprocessor/TransactionStateCache.java

```diff
@@ -85,7 +86,8 @@ private void tryInit() {
     try {
       Configuration conf = getSnapshotConfiguration();
       if (conf != null) {
-        this.storage = new HDFSTransactionStateStorage(conf, new SnapshotCodecProvider(conf));
+        this.storage = new HDFSTransactionStateStorage(conf, new SnapshotCodecProvider(conf),
+            new TxMetricsCollector());
```

Comment:
This class is only used by the TransactionProcessor coprocessor to load saved tx snapshots from HDFS so that we can clean up data on flush and compaction.  It's not involved in the remote call handling path, so it has no relevant metrics.  So I used the no-op metrics collector implementation here.  That was by design.

I can add a comment to clarify why this is.